### PR TITLE
Fix check test and add new ID scalar type that works right

### PR DESCRIPTION
**Pull request by rocktavious**

### CODE REVIEW

File: check.go [hunk 1]
```diff
@@ -145,10 +145,10 @@ type CheckUpdateInput struct {
 	Name     string        `json:"name,omitempty"`
 	Enabled  *bool         `json:"enabled,omitempty"`
 	EnableOn *iso8601.Time `json:"enableOn,omitempty"`
-	Category interface{}   `json:"categoryId,omitempty"`
-	Level    interface{}   `json:"levelId,omitempty"`
-	Owner    interface{}   `json:"ownerId,omitempty"`
-	Filter   interface{}   `json:"filterId,omitempty"`
+	Category interface{}   `json:"categoryId"`
```
Comment (eochoa260) [inline check.go:148]: So we'd have to change everything to an interface type?

File: scalar_test.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import (
+	"encoding/json"
+	"github.com/hasura/go-graphql-client"
+	"github.com/opslevel/opslevel-go/v2023"
+	"github.com/rocktavious/autopilot/v2022"
+	"testing"
+)
+
+type TestInput struct {
+	Key1 opslevel.ID  `json:"key1"`
```
Comment (eochoa260) [inline scalar_test.go:12]: Is there a reason these are `opslevel.ID` instead of `interface{}` like the update struct?

File: scalar_test.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		Key3: &id2,
+		Key4: &id2,
+	}
+	// Act
+	buf1, err1 := json.Marshal(one)
+	buf2, err2 := json.Marshal(two)
+	buf3, err3 := json.Marshal(three)
+	// Assert
+	autopilot.Ok(t, err1)
+	autopilot.Equals(t, `{"key1":"","key3":null}`, string(buf1))
```
Comment (eochoa260) [inline scalar_test.go:41]: This is awesome! 🎉

File: check.go (hunk 1 above)
Comment (rocktavious) [reply check.go:148]: well no.  This was a change from your branch.  If we are good with this new scalar type we'd change everything to an `opslevel.ID`

File: scalar_test.go (hunk 2 above)
Comment (rocktavious) [reply scalar_test.go:12]: The updated struct was a change from your branch that I was based on when i made this change.  Ideally if we are good with this we'll use `opslevel.ID` everywhere from here on out

**Review by eochoa260 — APPROVED** [review]

**Pull request merged**